content/docs/rfcs/70/README.md

```diff
@@ -0,0 +1,236 @@
+The proposal includes [Python and Rust implementations](https://github.com/sourav1547/htadkg).
+
+The DKG suggested makes assumes the existence of a PKI. 
+In case of requiring removing such assumption, one can replace the VSS scheme with the [Alhaddad et al.](https://eprint.iacr.org/2021/118) at the price of increasing the complexity.
+
+The output of the DKG may be an integer (modulo a prime), 
+meaning that one should apply a KDF to that output 
+in order to obtain a result which could be used as an input for the double ratchet.
+
+One observes that using an ADKG allows a set of users, 
+which want to define a group chat, 
+defining a common secret key which will be used as a root key for the double ratchet. 
+Using an ADKG defines a room key, 
+which essentially defines the group itself.
+
+
+This approach share similarities with the point of view of [Farcaster](https://github.com/farcasterxyz/protocol/discussions/99).
+
+Once the double ratchet is initialized, 
+the communication in this group is 1-to-1, 
+meaning that group member C cannot see the messages between group members A and B. 
+The fact of defining a room key makes impossible for outsiders to communicate with group members if the latter are not willing to.
+
+## n-to-n version
+
+Using the above approach leads to a situation where a group of users can set a group for 1-to-1 messages,
+meaning that any group member external to a communication between any other two members will not be able to read the contents of the messages.
+
+An approach to generalize this situation to the setting of a group of users exchanging messages without any kind of restriction is using asynchronous ratcheting trees, as suggested in the proposal from [Cohn-Gordon et al.](https://eprint.iacr.org/2017/666) where a group of people can derive a shared secret key even in the event of if no two users are ever online at the same time. 
+The proposal suggested provides both forward secrecy and post-compromise security. 
+The shared key can be then used in any symmetric encryption scheme, such as AES256.
+
+# Privacy and Security Considerations
+
+- For the information retrieval, the algorithm MUST include a access control mechanisms to restrict who can call the set and get functions.
+- One SHOULD include event logs to track changes in public keys.
+- The curve X448 MUST be chosen as the elliptic curve, since it offers a higher security level: 224-bit security instead of the 128-bit security provided by X25519.
+- Concerning the hardness of the ADKG, the proposal lies on the Discrete Logarithm assumption.
+
+# Copyright
+
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
+
+# References
+
+- https://rfc.vac.dev/spec/20/
+- https://signal.org/docs/specifications/x3dh/
+- https://signal.org/docs/specifications/doubleratchet/
+- https://eprint.iacr.org/2022/1389
+- https://github.com/sourav1547/htadkg
+- https://github.com/farcasterxyz/protocol/discussions/99
```

Comment:
How was the work here used? I don't think it was explicitly mentioned in the doc.